### PR TITLE
Allow configuring if Postgres structure dumps include comments

### LINKED ISSUE #44498: Column comments set with `add_column` are now stripped from PostgreSQL structure dumps
Commit https://github.com/rails/rails/commit/d5745607463ef72adb6a34e3ecc9b3820484a33f adds the `--no-comment` pg_dump option to the `structure_dump` database task. The linked issues suggest that the motivation for this was that setting comments on PostgreSQL extensions such as `plpgsql` requires elevated database permissions.

However, this breaks the [documented](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L559-L560) functionality of setting a `comment:` on database columns. While the comment will be added to the live database during the migration, it will no longer be emitted as part of the `structure.sql` dump, causing the committed structure to diverge from the schema in the database.

### Steps to reproduce

Add a column with a comment in a migration, then update the database structure dump.

```ruby
add_column(:users, :some_column, :integer, comment: 'the description of the column')
```

### Expected behavior

The comment is emitted in `structure.sql`, in such a way that it will be restored to the database as part of loading the schema.

```sql
--
-- Name: COLUMN users.some_column; Type: COMMENT; Schema: public; Owner: -
--

COMMENT ON COLUMN public.users.some_column IS 'the description of the column';
```

### Actual behavior

The comment is not emitted.

### System configuration
**Rails version**: [`main`](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/tasks/postgresql_database_tasks.rb#L61)

### PR DESCRIPTION
**Pull request by ghiculescu**
Introduces `ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.comments_in_structure_dump` to allow users to determine if `structure.sql` will include custom comments on tables or columns.

The default is true, meaning comments will be included. This is because prior to https://github.com/rails/rails/pull/43216, comments were included (and there were other features like `add_column comment:` that rely on this), and I think now we have a configuration option that is the more correct default.

Fixes https://github.com/rails/rails/issues/44498

### CODE REVIEW

**Review by eileencodes — COMMENTED** [review]
I understand why you made this the default but I'm concerned that pre-7 had comments, 7 added `--no-comments` and now we want to go back to comments in 7.1+.  I'm not a postgres user so I'm not entirely sure how disruptive this is. Thoughts?

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+*   Allow configuring if Postgres structure dumps include comments
```
[the remark this answers is not in the document]
Comment (ghiculescu) [reply activerecord/CHANGELOG.md:1]: My thinking is that disabling comments by default in 7.0 was a mistake. One way to minimise the amount of time that the behaviour is "wrong" could be to backport this to `7-0-stable`.

In terms of how disruptive this is:

- If you rely on comments on columns (anecdotally I haven't seen this much in the wild and we've used postgres in production for 10 years, *but* at least 1 user did report it), there's now no way to get them in your structure dump. I don't think this will break anyone's app, but those comments are presumably useful so we shouldn't remove them on your behalf.
- The issue described in https://github.com/rails/rails/issues/43107 would only happen if you're using an extension and you're running CI in an environment that lacks superuser (or I guess if you're loading a development environment from a structure dump in a restricted environment?). Using extensions is pretty common, but I'd hope that whatever CI you're running in you can add superuser if you need it. Again, and at least 1 user has reported it.

In short, I don't think it's a huge disruption because there's workarounds either way. But we should try and make the issue go away.

[the remark this answers is not in the document]
Comment (matthewd) [reply activerecord/CHANGELOG.md:1]: > we just revert https://github.com/rails/rails/pull/43216 and push people to https://github.com/rails/rails/issues/43107#issuecomment-906573620

Yeah, that's my inclination. There are already multiple valid workarounds here, so no need for us to invent an additional almost-but-not-quite similar means of setting that option. (I believe the upstream-recommended workaround would be to run psql in default mode, where it will not abort upon the ignorable comment-setting error, and instead continue happily.)

This is really a Postgres bug, though I don't know if it's been discussed upstream there. Comments on extensions get created as a side-effect of adding an extension, which by any reasonable definition means they belong to the extension, and therefore [just like the extension's schema, tables, functions, etc] should not be separately [and redundantly] dumped by pg_dump.

[the remark this answers is not in the document]
Comment (ghiculescu) [reply activerecord/CHANGELOG.md:1]: Okay, I can make a PR for that.

[the remark this answers is not in the document]
Comment (ghiculescu) [reply activerecord/CHANGELOG.md:1]: https://github.com/rails/rails/pull/44633
